src/sherpa_ai/connectors/chroma_vector_store.py

```diff
@@ -0,0 +1,121 @@
+import uuid
+
+import chromadb
+from chromadb.utils import embedding_functions
+from langchain.docstore.document import Document
+from langchain.text_splitter import CharacterTextSplitter
+
+import sherpa_ai.config as cfg
+
+
+class ChromaVectorStore:
+    """
+    A class used to represent a Chroma Vector Store.
+
+    This class provides methods to create a Chroma Vector Store from texts or from an existing store,
+    split file text, and perform a similarity search.
+
+    ...
+
+    Attributes
+    ----------
+    db : chromadb.PersistentClient
+        a persistent client to interact with the ChromaDB
+
+    Methods
+    -------
+    chroma_from_texts(texts, embedding, meta_datas)
+        Class method to create a Chroma Vector Store from given texts.
+    chroma_from_existing(embedding)
+        Class method to create a Chroma Vector Store from an existing store.
+    file_text_splitter(data, meta_data)
+        Class method to split file text into chunks.
+    similarity_search(query, session_id)
+        Method to perform a similarity search in the Chroma Vector Store.
+    """
+
+    def __init__(self, db, path="./db") -> None:
+        self.db = db
+        self.path = path
+
+    @classmethod
+    def chroma_from_texts(
+        cls,
+        texts,
+        embedding=embedding_functions.OpenAIEmbeddingFunction(
+            model_name="text-embedding-ada-002"
+        ),
+        meta_datas=None,
+        path="./db",
+    ):
+        embeded_data = embedding(texts)
+        meta_datas = [] if meta_datas is None else meta_datas
+        client = chromadb.PersistentClient(path=path)
+        db = client.get_or_create_collection(
+            name=cfg.INDEX_NAME_FILE_STORAGE, embedding_function=embedding
```

Comment:
I think we've discussed changing this as a parameter to the ChromaVectorStore with a default value. So  this can be easily configured with Hydra